code/game/machinery/doors/airlock.dm

```diff
@@ -1508,6 +1512,9 @@ About the new airlock wires panel:
 /obj/machinery/door/airlock/enable_AI_control(var/bypass = FALSE)
 	if(bypass)
 		aiControlDisabled = 2
-	else 
+	else
 		aiControlDisabled = 0
-	
\ No newline at end of file
+
+/obj/machinery/door/airlock/tackled(mob/living/carbon/human/user)
+	if(user.wear_id && istype(user.wear_id, /obj/item/weapon/card/emag))
```

Comment:
I'm pretty sure monkeys are able to tackle, not just humans. Trying to access the monkey's `wear_id` here would result in a runtime error: you should check whether the user is actually a human first.
Also, `istype` will return false if the input passed to it is null, so you don't need the first `user.wear_id` check if you're going to use `istype`.

Reply:
Doesn't 'tackled(mob/living/carbon/human/user)' preclude this proc from firing for any mob except for a human?

Reply:
Nope. It just changes the type with which the `user` argument is declared. The actual value that will be stored in that var can be anything.

Reply:
Thanks, fixed in latest commit